iothub/device/src/Common/Security/SharedAccessSignature.cs

```diff
@@ -205,5 +206,26 @@ internal static string Sign(byte[] key, string value)
 
             return parsedFields;
         }
+
+        // N.B. This allows for string comparisons without a potential security vulnerability.
+        // Without this it could be possible to attempt different values and determine length
+        // and maybe even entire string through brute force atttempts.
+        //
+        // This is also a function that can be removed later once the C# version being used
+        // supports https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.cryptographicoperations.fixedtimeequals?view=net-5.0
+        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
+        private static bool FixedTimeEquals(string left, string right)
+        {
+            if (left.Length != right.Length)
+            {
+                return false;
+            }
+            var result = 0;
+            for (var i = 0; i < left.Length; i++)
+            {
+                result |= left[i] ^ right[i];
```

Comment:
I'd recommend we replace the xor with a subtraction. We'd then follow the same implementation that dotnet runtime uses: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptographicOperations.cs#L31

Reply:
This is something that can definitely live in a common helper, we can propose that change if you'd like.

Reply:
Yeah, if there is good location that this should live in, I can move it there or it can be done after this PR. Whichever is fine with me :)